app/Infrastructure/Controllers/GetsWalletCryptocurrenciesController.php

```diff
@@ -2,35 +2,26 @@
 
 namespace App\Infrastructure\Controllers;
 
-use App\Domain\DataSources\WalletDataSource;
+use App\Application\Services\WalletCryptocurrenciesService;
 use App\Validators\WalletIdValidator;
 use Illuminate\Http\JsonResponse;
 use Illuminate\Http\Response;
 use Illuminate\Routing\Controller as BaseController;
-use Illuminate\Support\Facades\Cache;
 
 class GetsWalletCryptocurrenciesController extends BaseController
 {
-    private WalletDataSource $walletDataSource;
-
-    public function __construct(WalletDataSource $walletDataSource)
-    {
-        $this->walletDataSource = $walletDataSource;
-    }
-    public function __invoke($wallet_id, WalletIdValidator $walletIdValidator): JsonResponse
-    {
+    public function __invoke(
+        $wallet_id,
+        WalletIdValidator $walletIdValidator,
+        WalletCryptocurrenciesService $walletCryptocurrenciesService
+    ): JsonResponse {
         if (!$walletIdValidator->validateWalletId($wallet_id)) {
             return response()->json(['error' => 'Bad Request',
```

Comment:
For being more readable
```suggestion
            return response()->json(
                [
                    'error' => 'Bad Request',
                    'message' => $walletIdValidator->getMessage($wallet_id)
                ], Response::HTTP_BAD_REQUEST);
```